ncloud/resource_ncloud_sourcecommit_repository.go

```diff
@@ -0,0 +1,268 @@
+}
+
+func resourceNcloudSourceCommitRepository() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceNcloudSourceCommitRepositoryCreate,
+		ReadContext:   resourceNcloudSourceCommitRepositoryRead,
+		UpdateContext: resourceNcloudSourceCommitRepositoryUpdate,
+		DeleteContext: resourceNcloudSourceCommitRepositoryDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(DefaultTimeout),
+			Read:   schema.DefaultTimeout(DefaultTimeout),
+			Update: schema.DefaultTimeout(DefaultTimeout),
+			Delete: schema.DefaultTimeout(DefaultTimeout),
+		},
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				ValidateDiagFunc: ToDiagFunc(validation.StringLenBetween(1, 100)),
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ValidateDiagFunc: ToDiagFunc(validation.StringLenBetween(0, 500)),
+			},
+			"creator": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"git_https_url": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"git_ssh_url": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"filesafer": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func resourceNcloudSourceCommitRepositoryCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*ProviderConfig)
+
+	reqParams := &sourcecommit.CreateRepository{
+		Name:        ncloud.String(d.Get("name").(string)),
+		Description: StringPtrOrNil(d.GetOk("description")),
+	}
+
+	if fileSafer, ok := d.GetOk("filesafer"); ok {
+		reqParams.Linked = &sourcecommit.CreateRepositoryLinked{
+			FileSafer: BoolPtrOrNil(fileSafer, ok),
+		}
+	}
+
+	logCommonRequest("resourceNcloudSourceCommitRepositoryCreate", reqParams)
+	resp, err := config.Client.sourcecommit.V1Api.CreateRepository(ctx, reqParams)
+	logCommonResponse("resourceNcloudSourceCommitRepositoryCreate", GetCommonResponse(nil))
+	var diags diag.Diagnostics
+
+	if err != nil {
+		logErrorResponse("resourceNcloudSourceCommitRepositoryCreate", err, reqParams)
+
+		diags = append(diags, diag.Diagnostic{
+			Severity: diag.Error,
+			Summary:  "Fail to create repository",
+			Detail:   err.Error(),
+		})
+		return diags
+	}
+
+	logResponse("resourceNcloudSourceCommitRepositoryCreate", resp)
+
+	name := ncloud.StringValue(reqParams.Name)
+
+	if err := waitForSourceCommitRepositoryActive(ctx, d, config, name); err != nil {
+
+		name := d.Get("name").(string)
+
+		diags := append(diag.FromErr(err), diag.Diagnostic{
+			Severity: diag.Error,
+			Summary:  "Unable to search repository",
+			Detail:   fmt.Sprintf("Unable to search repository - detail , name : (%s)", name),
+		})
+		return diags
+	}
```

Comment:
```suggestion

	name := ncloud.StringValue(reqParams.Name)

	if err := waitForSourceCommitRepositoryActive(ctx, d, config, name); err != nil {
		diags := append(diag.FromErr(err), diag.Diagnostic{
			Severity: diag.Error,
			Summary:  "Unable to search repository",
			Detail:   fmt.Sprintf("Unable to search repository - detail , name : (%s)", name),
		})
		return diags
	}
```

`name`은 위 `name` 변수 재사용하면 될 것 같습니다~

Reply:
넵 수정하였습니다~!